dockstore-webservice/src/main/java/io/dockstore/webservice/resources/TokenResource.java

```diff
@@ -321,34 +335,52 @@ public Token addGoogleToken(@ApiParam(hidden = true) @Auth Optional<User> authUs
         Token dockstoreToken = null;
         Token googleToken = null;
         String googleLoginName = userinfo.getEmail();
-        User user = userDAO.findByUsername(googleLoginName);
-
-        List<Token> googleByUsername = tokenDAO.findTokenByUsername(userinfo.getEmail(), TokenType.GOOGLE_COM);
-        if (user == null && !authUser.isPresent() && googleByUsername.isEmpty()) {
-            user = new User();
-            // Pull user information from Google
-            user.setUsername(userinfo.getEmail());
-            userID = userDAO.create(user);
-
-            // CREATE DOCKSTORE TOKEN
-            dockstoreToken = createDockstoreToken(userID, googleLoginName);
+        User user;
+        User googleUser;
+        googleUser = userDAO.findByUsername(googleLoginName);
+        List<Token> tokensByGoogleUsername = tokenDAO.findTokenByUsername(userinfo.getEmail(), TokenType.GOOGLE_COM);
+        if (authUser.isPresent()) {
```

Comment:
Suggestion, but feel free to ignore: if this whole if/else block were a function, I think it would be easier to read. I was looking at line 397 to see if userID was always initialized, and I had to check that each if/else ended up setting the variable. Whereas if you had a function that returned a userID, it would be clear.

The problem is that in one branch you are also creating a dockstore token, so you can't return just the userID, so then you have return an instance of a new class, or a Pair, or something like that, so maybe it's not worth it.

Reply:
Yes, i was considering doing something that like in the beginning but decided against it because of the dockstore token.  There's an IDE error or at least a warning when userId is possibly not initialized, so i wasn't really worried about that.